Makefile

```diff
@@ -144,6 +144,42 @@ ifeq ($(UNAME_M),$(filter $(UNAME_M),x86_64 i686 amd64))
 			CXXFLAGS += -mavx2
 		endif
 
+		AVX512F_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512F')
+		ifneq (,$(AVX512F_M))
+			CFLAGS   += -mavx512f
+			CXXFLAGS += -mavx512f
+		endif
+
+		AVX512DQ_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512DQ')
+		ifneq (,$(AVX512DQ_M))
+			CFLAGS   += -mavx512dq
+			CXXFLAGS += -mavx512dq
+		endif
+
+		AVX512CD_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512CD')
+		ifneq (,$(AVX512CD_M))
+			CFLAGS   += -mavx512cd
+			CXXFLAGS += -mavx512cd
+		endif
+
+		AVX512CD_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512BW')
+		ifneq (,$(AVX512BW_M))
+			CFLAGS   += -mavx512bw
+			CXXFLAGS += -mavx512bw
+		endif
+
+		AVX512VL_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512VL')
+		ifneq (,$(AVX512VL_M))
+			CFLAGS   += -mavx512vl
+			CXXFLAGS += -mavx512vl
+		endif
```

Comment:
I suggest combining them into one block, where we check just for AVX512F, but then add flags related to all 5 AVX-512 subsets (F, CD, VL, DQ, BW). I don't believe Xeon Phi existence is good enough reason check all of these subsets individually. I never suggested checking them individually.

Table at https://en.wikipedia.org/wiki/AVX-512#CPUs_with_AVX-512 explains why.

Let's have a simpler code. If there will be mainstream CPU in the future not fitting into the assumption (F presence means presence of CD, VL, DQ and BW too), then we can start playing again with checking those flags individually.